airflow/gcp/operators/bigquery.py

```diff
@@ -594,6 +594,12 @@ def on_kill(self):
             self.log.info('Cancelling running query')
             self.bq_cursor.cancel_query()
 
+    @classmethod
+    def get_serialized_fields(cls):
+        """Serialized BigQueryOperator contain exactly these fields."""
+        cls._serialized_fields = frozenset(super().get_serialized_fields() | {"sql"})
+        return cls._serialized_fields
```

Comment:
These should have the `if not cls._serialied_fields` guard too. It won't break without it (as a set won't have the value more than once)

Reply:
The problem is when calling `super().get_serialized_fields()` it would set `cls._serialized_fields` from the BaseOperator, hence it would not have `{"sql"}` in it